odoo/addons/base/models/res_currency.py

```diff
@@ -52,6 +52,20 @@ class Currency(models.Model):
         ('rounding_gt_zero', 'CHECK (rounding>0)', 'The rounding factor must be greater than 0!')
     ]
 
+    def write(self, vals):
+        res = super().write(vals)
+        if 'active' not in vals:
+            return res
+        # Automatically activate group_multi_currency if there is more than 1 active currency, deactivate it otherwise
+        group_user = self.env.ref('base.group_user', raise_if_not_found=False)
+        group_mc = self.env.ref('base.group_multi_currency', raise_if_not_found=False)
+        active_currency_count = self.search_count([('active', '=', True)])
+        if group_user and group_mc and active_currency_count > 1:
+            group_user.write({'implied_ids': [(4, group_mc.id)]})
+        elif group_user and group_mc and active_currency_count <= 1:
+            group_user.write({'implied_ids': [(3, group_mc.id)]})
```

Comment:
I'd avoid re-computing the group hierarchy when no change is necessary (takes a a while on large databases), as there will usually be only 1 such transition in the whole lifetime of a database, and never again.

Also, the active-record syntax can be more readable:

```suggestion
        if group_user and group_mc and active_currency_count > 1 and group_mc not in group_user.implied_ids:
            group_user.implied_ids |= group_mc
        elif group_user and group_mc and active_currency_count <= 1 and group_mc in group_user.implied_ids:
            group_user.implied_ids -= group_mc
```

Reply:
Thanks @odony. I have modified according to your suggestions.

I still have the issue of requiring an explicit reload. It is caused by f6c5757